### DISCUSSION TITLE
Predictive Address Search

### DISCUSSION THREAD

**Discussion by TheDreamFactoryNZ** [Budibase/budibase#14312] [Help]
Given Budibase still doesn't allow external scripts, I'm unable to load Google Maps' Place Autocomplete tool (this [here](https://developers.google.com/maps/documentation/javascript/examples/places-autocomplete?csw=1))

I want a form field where people can enter in an address and then a list of suggested addresses appear. The tool above allows for limiting the search to specific countries etc too. Any existing examples of this or other viable alternatives?

Cheers

**Comment by ConorWebb96** [comment]
Hey @TheDreamFactoryNZ,

You may want to look into custom plugins I believe there is an autocomplete plugin [there](https://github.com/Budibase/plugins).

**Reply by TheDreamFactoryNZ** [reply]
Hi Conor,

What an awesome plugin. Thanks for sharing that with me @ConorWebb96 - I must have missed it the first time I was trawling through there.

Another question for you: I've gotten this working on my own host except it would appear the Google Maps API is blocked by Budibase's CSP settings. So it would appear even with this plugin, predictive address search is prohibited as there's nowhere to get addresses from in the first place.

Even when I attempted to manually set CSP policies via Nginx it would appear there's still something blocking the Google Maps API. [I've found a comment](https://github.com/Budibase/budibase/discussions/9121#discussioncomment-4581727) from @mike12345567 stating I have to build my own version of the budibase/proxy image?

Is this still the case or is there a better way to adjust CSP settings?

**Reply by A3DataSolutions** [reply]
@TheDreamFactoryNZ 

I'm working on a couple custom plugins (datasource and component) just for this functionality! Although datasource plugins are also not supported on budibase cloud but might be a workaround with API datasource. 


https://github.com/user-attachments/assets/b02aa4fe-a010-45ff-a516-1c6c352e4624

**Reply by TheDreamFactoryNZ** [reply]
@A3DataSolutions That looks great! How are you able to achieve that though, I would have thought Budibase would have just blocked your efforts to tap into external data sources?

**Reply by A3DataSolutions** [reply]
So this is on a self hosted instance using two custom plugins, but you can do the same on budibase cloud using the REST API connector, a bit of leg work but definitely possible.
